v1/errors/sentry.go

```diff
@@ -0,0 +1,115 @@
+	}
+)
+
+// sentryEventModifier implements the sentry.EventModifier interface.
+type sentryEventModifier struct {
+	tags  map[string]string
+	panic bool
+}
+
+func (m *sentryEventModifier) ApplyToEvent(event *sentry.Event, _ *sentry.EventHint) *sentry.Event {
+	// Clean stack traces by filtering non-relevant frames
+	for i := range event.Exception {
+		event.Exception[i].Stacktrace.Frames = filterStackFrames(event.Exception[i].Stacktrace.Frames)
+	}
+	for i := range event.Threads {
+		event.Threads[i].Stacktrace.Frames = filterStackFrames(event.Threads[i].Stacktrace.Frames)
+	}
+
+	// Add tags extracted from the original log record context
+	event.Tags = m.tags
+
+	// If the event has been created following a panic, flag it as crashed
+	if m.panic && len(event.Threads) == 1 {
+		event.Threads[0].Crashed = true
+	}
+
+	return event
+}
+
+// sentryEventFromPanic returns a sentryEventModifier instance containing tags extracted from a log record's context.
+func sentryEventFromLogRecord(rec *log15.Record) *sentryEventModifier {
+	var s = sentryEventModifier{tags: make(map[string]string)}
+
+	// Extract context key/value pairs from the log record's context,
+	// which is stored as a string slice of key/value pairs.
+	for i := 0; i < len(rec.Ctx); i += 2 {
+		k, v := rec.Ctx[i], rec.Ctx[i+1]
+		s.tags[fmt.Sprint(k)] = fmt.Sprint(v)
+	}
+
+	return &s
+}
+
+// sentryEventFromPanic returns a sentryEventModifier instance indicating a crash.
+func sentryEventFromPanic(_ interface{}) *sentryEventModifier {
+	return &sentryEventModifier{
+		tags:  make(map[string]string),
+		panic: true,
+	}
+}
+
+// filterStackFrames filters out all frames related to internal packages.
+func filterStackFrames(frames []sentry.Frame) []sentry.Frame {
+	var filteredFrames = make([]sentry.Frame, 0)
+
+nextFrame:
+	for _, frame := range frames {
+		for _, pkg := range internalPackages {
+			if strings.HasPrefix(frame.Module, pkg) {
+				continue nextFrame
+			}
+		}
+		filteredFrames = append(filteredFrames, frame)
+	}
+
+	return filteredFrames
+}
+
+// SentryTestTransport is an implementation of the sentry.Transport interface for testing purposes.
+type SentryTestTransport struct {
+	mu        sync.Mutex
+	events    []*sentry.Event
+	lastEvent *sentry.Event
+}
+
+// Configure is a no-op for SentryTestTransport.
+func (t *SentryTestTransport) Configure(options sentry.ClientOptions) {}
+
+// SendEvent assembles a new packet out of `Event` and sends it to remote server.
+func (t *SentryTestTransport) SendEvent(event *sentry.Event) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	t.events = append(t.events, event)
+	t.lastEvent = event
+}
+
+// Flush is a no-op for SentryTestTransport. It always returns true immediately.
+func (t *SentryTestTransport) Flush(timeout time.Duration) bool {
+	return true
+}
+
+// Events returns a list of events received by the Transport.
+func (t *SentryTestTransport) Events() []*sentry.Event {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	return t.events
```

Comment:
here you only read the value. Shouldn't be:
```Golang
t.mu.RLock()
defer t.mu.RUnlock()
```

Reply:
Another point here. In the result of this function you are returning a slice of pointer.
In an asynchronous context: Careful if you will read/write in an element of this slice, because it's not a copy just a reference to this object.
If you know what I mean

Reply:
To be totally transparent, this code has been "borrowed" [from here](https://github.com/getsentry/sentry-go/blob/master/mocks_test.go) since it's not possible to use code located in `_test.go` files from an external package. I'm not sure we're subject to performance issues in this context 😉

Reply:
Yes sure I agreed :) It's just a good practice to `RLock` on a write.
For the second point It was just a "FWI"  for your futures test ;)

Reply:
I don't know what *FWI* means, but thanks I'll take note :)

Reply:
"FYI" Sometime a bit confused with my `fr` and `ch-fr` keybord 😂